### PR TITLE
[GUILD] Rewrite `SplitButton` in TS

### PR DESCRIPTION
**Pull request by KristofColpaert**
### Description

PR for the Typescript rewrite of the `SplitButton`.

### Manual check 

Please check Storybook.

### CODE REVIEW

File: src/components/utils/isReactElement.ts [hunk 1]
```diff
@@ -0,0 +1,6 @@
+import { ReactElement } from 'react';
+
+export default (node: unknown): node is ReactElement => {
```
Comment (eniskraasniqi1) [inline src/components/utils/isReactElement.ts:3]: Q: Doesn't React.isValidElement already check this ?

File: src/components/splitButton/SplitButton.tsx [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+import { Menu } from '../menu';
+import Popover from '../popover';
+import { IconChevronDownSmallOutline } from '@teamleader/ui-icons';
+import { BoxProps } from '../box/Box';
+import { GenericComponent } from '../../@types/types';
+import isReactElement from '../utils/isReactElement';
+
+export type Level = 'primary' | 'secondary' | 'destructive';
+export type Size = 'small' | 'medium' | 'large';
+interface SplitButtonProps extends Omit<BoxProps, 'children' | 'size'> {
```
Comment (eniskraasniqi1) [inline src/components/splitButton/SplitButton.tsx:14]: size is redundant here I think

File: src/components/splitButton/SplitButton.tsx [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+  onSecondaryButtonClick,
+  disabled,
+  ...others
+}) => {
+  const [buttonLabel, setButtonLabel] = useState<string>();
+  const [popoverActive, setPopoverActive] = useState<boolean>(false);
+
+  const popoverAnchorEl = useRef<HTMLElement | null>(null);
+
+  useEffect(() => {
```
Comment (stefaandevylder) [inline src/components/splitButton/SplitButton.tsx:43]: Is it really needed to do a whole side effect for this?

File: src/components/utils/isReactElement.ts (hunk 1 above)
Comment (jelledc) [reply src/components/utils/isReactElement.ts:3]: @eniskraasniqi1 This is basically a type guard, which serves a different purpose than `isValidElement`. I don't think that one can be used in the same way like this one.

Comment (KristofColpaert) [reply src/components/utils/isReactElement.ts:3]: Yes, you're right. Thanks for the advise! I've updated the PR ;)

File: src/components/splitButton/SplitButton.tsx (hunk 2 above)
Comment (KristofColpaert) [reply src/components/splitButton/SplitButton.tsx:14]: Size could be in BoxProps HTMLProps ;)

**Review by qubis741 — APPROVED** [review]

**Review by stefaandevylder — DISMISSED** [review]

**Review by qubis741 — APPROVED** [review]

**Comment by eniskraasniqi1** [comment]
MC`ing ⏳

**Review by eniskraasniqi1 — APPROVED** [review]
Manual check looks good ✅

**Pull request merged**